lib/gui/lvgl/lvgl.c

```diff
@@ -69,6 +69,19 @@ static void lvgl_log(lv_log_level_t level, const char *file, uint32_t line,
 
 static int lvgl_allocate_rendering_buffers(lv_disp_drv_t *disp_drv)
 {
+	struct display_capabilities cap;
+	struct device *display_dev = (struct device *)disp_drv->user_data;
+
+	display_get_capabilities(display_dev, &cap);
+
+	__ASSERT(cap.x_resolution <= CONFIG_LVGL_HOR_RES,
```

Comment:
I suggest to treat it as error and not just ASSERT which is not enabled per default.

Reply:
Start to wonder if this should be handled as an assert and/or error at all.
In the end there is no issue with only partially using the display.